precompiles/xvm/src/lib.rs

```diff
@@ -95,18 +95,40 @@ where
         let call_to = input.read::<Bytes>()?.0;
         let call_input = input.read::<Bytes>()?.0;
 
-        // Build call with origin.
-        let origin = Some(R::AddressMapping::into_account_id(handle.context().caller)).into();
-        let call = pallet_xvm::Call::<R>::xvm_call {
-            context,
-            to: call_to,
-            input: call_input,
-        };
-
-        // Dispatch a call.
-        // The underlying logic will handle updating used EVM gas based on the weight of the executed call.
-        RuntimeHelper::<R>::try_dispatch(handle, origin, call)?;
-
-        Ok(succeed(EvmDataWriter::new().write(true).build()))
+        let from = R::AddressMapping::into_account_id(handle.context().caller);
+        match &pallet_xvm::Pallet::<R>::xvm_bare_call(context, from, call_to, call_input) {
+            result @ Ok(success) => {
+                log::trace!(
+                    target: "xvm-precompile::xvm_call",
+                    "success: {:?}", success
+                );
+
+                Ok(succeed(
+                    EvmDataWriter::new()
+                        .write(true)
+                        .write(pallet_xvm::consumed_weight(&result))
+                        .write(Bytes(success.output().clone())) // FIXME redundant clone
```

Comment:
Perhaps it's better to leave a **TODO** given we have those in this repository. We don't have any **FIXME**.